qa/rpc-tests/test_framework/cashlib/cashlib.py

```diff
@@ -0,0 +1,152 @@
+from ctypes import *
+from binascii import hexlify, unhexlify
+import pdb
+import hashlib
+import decimal
+
+SIGHASH_ALL = 1
+SIGHASH_NONE = 2
+SIGHASH_SINGLE = 3
+SIGHASH_FORKID = 0x40
+SIGHASH_ANYONECANPAY = 0x80
+
+BCH = 100000000
+
+cashlib = None
+
+class Error(BaseException):
+    pass
+
+def init(libbitcoincashfile=None):
+    global cashlib
+    if libbitcoincashfile is None:
+        libbitcoincashfile = "libbitcoincash.so"
+        try:
+            cashlib = CDLL(libbitcoincashfile)
+        except OSError:
+            import os
+            dir_path = os.path.dirname(os.path.realpath(__file__))
+            cashlib = CDLL(dir_path + os.sep + libbitcoincashfile)
+    else:
+        cashlib = CDLL(libbitcoincashfile)
+    if cashlib is None:
+        raise Error("Cannot find %s shared library", libbitcoincashfile)
+
+
+# Serialization/deserialization tools
+def sha256(s):
```

Comment:
this function is copy/pasted from `nodemessages.py`, use the on already define rather than add code duplication.

Reply:
cashlib is supposed to stand alone, not be tightly integrated into the test_framework.  These APIs will likely be replaced with their C versions so that they can be accessed from other languages, and be high performance (although python likely uses a c implementation of hashlib, its probably not optimized assembly like bitcoin has).  Right now I implemented them in python via copy-paste so that I can gauge their usefulness as part of the cashlib API.